src/ovs/aca_ovs_l3_programmer.cpp

```diff
@@ -829,6 +829,107 @@ int ACA_OVS_L3_Programmer::create_or_update_router(RouterConfiguration &current_
             new_routing_rule_entry.next_hop_mac =
                     current_routing_rule.routing_rule_extra_info().next_hop_mac();
 
+            auto remote_host_ip = "";
+            ulong culminative_dataplane_programming_time = 0;
+            for (auto each_neighbor : parsed_struct.neighbor_states()) {
+              NeighborConfiguration current_NeighborConfiguration1 =
+                      each_neighbor.second.configuration();
+              ACA_LOG_INFO("current_NeighborConfiguration.host_ip_address(): %s \n",
+                            current_NeighborConfiguration1.host_ip_address().c_str());
+              for (int y = 0; y < current_NeighborConfiguration1.fixed_ips_size(); y++) {
+                ACA_LOG_INFO("current_NeighborConfiguration.fixed_ips(%d): neighbor_type: %d, subnet_id %s, ip_address %s \n",
+                              y, current_NeighborConfiguration1.fixed_ips(y).neighbor_type(),
+                              current_NeighborConfiguration1.fixed_ips(y)
+                                      .subnet_id()
+                                      .c_str(),
+                              current_NeighborConfiguration1.fixed_ips(y)
+                                      .ip_address()
+                                      .c_str());
+                ACA_LOG_INFO("current_routing_rule.next_hop_ip() %s\n",
+                              current_routing_rule.next_hop_ip().c_str());
+                auto current_fixed_ip = current_NeighborConfiguration1.fixed_ips(y);
+                string virtual_ip_address = current_fixed_ip.ip_address();
+                string virtual_mac_address =
+                        current_NeighborConfiguration1.mac_address();
+                string subnet_id_of_fixed_ip = current_fixed_ip.subnet_id();
+                string gw_mac;
+                uint dest_tunnel_id = 0;
+
+                if (strcmp(current_routing_rule.next_hop_ip().c_str(),
+                            current_fixed_ip.ip_address().c_str()) == 0) {
+                  auto subnet_iterator = parsed_struct.subnet_states().find(subnet_id_of_fixed_ip);
+                  if(subnet_iterator != parsed_struct.subnet_states().end()){
```

Comment:
space between 'if' and '(', and before '{' please

Reply:
added